src/transformers/modeling_utils.py

```diff
@@ -4001,6 +4001,11 @@ def _fix_key(key):
                 else:
                     _loaded_keys = loaded_keys
                 set_initialized_submodules(model, _loaded_keys)
+                # if we're about to tie the output embeds to the input embeds we don't need to init them
+                if hasattr(model.config, "tie_word_embeddings") and model.config.tie_word_embeddings:
+                    output_embeddings = model.get_output_embeddings()
+                    if output_embeddings is not None:
+                        output_embeddings._is_hf_initialized = True
```

Comment:
Very nice!
I think we can also add something like https://github.com/huggingface/transformers/blob/90224dd59e92e11d99b5b09be84d3fe7794636b9/src/transformers/modeling_utils.py#L1580

for 
```python
        if getattr(self.config, "is_encoder_decoder", False) and getattr(self.config, "tie_encoder_decoder", False):
            if hasattr(self, self.base_model_prefix):
                self = getattr(self, self.base_model_prefix)
            self._tie_encoder_decoder_weights(self.encoder, self.decoder, self.base_model_prefix)
```
which can be tested on MarianMT models but should come in another PR!